go/pfconfigdriver/structs.go

```diff
@@ -0,0 +1,41 @@
+package pfconfigdriver
+
+import (
+	"encoding/json"
+)
+
+// Interface for a pfconfig object. Not doing much now but it is there for future-proofing
+type PfconfigObject interface {
+}
+
+// pfconfig replies with the «struct» nested in an element key of a hash
+// ex: {"element":{"data1":"value1", "data2":"value2"}}
+// The structs are built to receive the value of element, so in order to have 2 stage decoding, this serves as a receiver for the pfconfig payload which then gets decoded into the right type
+type PfconfigElementResponse struct {
+	Element *json.RawMessage
+}
+
+// To be combined with another struct in order to give it a Type attribute common in PF
+type TypedConfig struct {
+	Type string `json:"type"`
+}
+
+// Represents the pf.conf general section
+type PfConfGeneral struct {
```

Comment:
I was thinking of it as pf.conf general as its not really tied to pfconfig but more to the structure of the file